js/controller/ContentController.js

```diff
@@ -85,28 +106,11 @@ app.controller('ContentController', function (Publisher, FeedResource, ItemResou
 
         item.keepUnread = !item.keepUnread;
     };
-
-    var getOrdering = function () {
-        var ordering = SettingsResource.get('oldestFirst');
-
-        if (self.isFeed()) {
-            var feed = FeedResource.getById($routeParams.id);
-            if (feed && feed.ordering === 1) {
-                ordering = true;
-            } else if (feed && feed.ordering === 2) {
-                ordering = false;
-            }
-        }
-
-        return ordering;
-    };
-
-    this.orderBy = function () {
-        if (getOrdering()) {
-            return 'id';
-        } else {
-            return '-id';
-        }
+    
+    this.sortIds = function(first, second) {
```

Comment:
This function replaces `this.orderBy`. `this.orderBy` would return the name of the field to use for sorting (adding a minus sign means to sort by that field in reverse order), and then used the default AngularJS comparison to sort on that field. That doesn't work here, since our IDs are strings, but we want to treat them as numbers. So, `this.sortIds` is a custom comparator that parses them as integers first.

Custom comparators are covered here: https://docs.angularjs.org/api/ng/filter/orderBy#using-a-custom-comparator